### PR TITLE
Integration test upon unresolvable Maven artifacts in BOMs

### LINKED ISSUE #1392: LinkageCheckerRule does not output ArtifactProblems from findBomClassPath
Output ArtifactProblems, if any, of ClassPathResult of findBomClassPath.

First, write a test to confirm the current behavior.

### PR DESCRIPTION
**Pull request by suztomo**
Closes #1392 , in which I raised question on logging artifact problems. Because the enforcer rule fails upon unresolvable artifacts, there's no need to log artifact problems.

### CODE REVIEW

File: enforcer-rules/src/it/bom-project-missing-artifact/verify.groovy [hunk 1]
```diff
@@ -0,0 +1,5 @@
+def buildLog = new File(basedir, "build.log").text
+
+assert buildLog.contains("Failed to collect dependency: "
+        +"[xerces:xerces-impl:jar:2.6.2 was not resolved. "
+        +"Dependency path: ant:ant:jar:1.6.2 (compile) > xerces:xerces-impl:jar:2.6.2 (compile?)")
```
[the remark this answers is not in the document]
Comment (elharo) [reply enforcer-rules/src/it/bom-project-missing-artifact/verify.groovy:5]: Is there a change that needs to be made to stop using that logger?

[the remark this answers is not in the document]
Comment (suztomo) [reply enforcer-rules/src/it/bom-project-missing-artifact/verify.groovy:5]: No need. The logger is currently not used for this.

[the remark this answers is not in the document]
Comment (suztomo) [reply enforcer-rules/src/it/bom-project-missing-artifact/verify.groovy:5]: I was wrong. There's unnecessary warning message.

[the remark this answers is not in the document]
Comment (suztomo) [reply enforcer-rules/src/it/bom-project-missing-artifact/verify.groovy:5]: The warning message ([example](https://gist.github.com/suztomo/0097c48ba5b61ab1140b9209f8de319c#file-gistfile1-txt-L8)) comes from enforcer rule plugin. It's not something we can fix at LinkageCheckerRule class.

Therefore, no need to stop using that logger.

**Pull request merged**